### PR TITLE
Added --control-id flag

### LINKED ISSUE #218: Discuss the use of STIG field `Rule ID` as `InSpec Control id`
The intent of the issue is to discuss the use of STIG field `Rule ID` as `InSpec Control id` rather than `Vul Id` that it is currently using.

1) Discuss the implication/benefits of the change.
2) Understand the scope of updates that has to be made to toolset and profiles.
3) Plan and create sub-tickets to update the elements. 
3) Strategize communication to users of the profiles and toolset about the update.

Notes from meeting with RMO re the STIG Data fields.

_**Vul ID**: This field will be depreciated in the near future. It is a remnant of the now retired VMS._

_**STIG ID**: This not of a stable format and can be modified at any release._

_**Rule ID**: This is the stable identifier to go with, since it relates to a database index of how the STIG is stored. 
Legacy Rule ID are of format `SV-86473r2_rule` and new format is `SV-204392r505924_rule`
The part of the Rule ID before `r` will be stable across revisions/versions.
ie `SV-86473` will be identical across generation of the same STIG.
However Rule ID will be different for RHEL7 to RHEL8 STIG._

### PR DESCRIPTION
**Pull request by wdower**
Added --control-id flag to xccdf2inspec to allow user to specify if they want the created profile to use legacy vuln IDs (ex. 'V-XXXXX') or rule IDs as the control IDs for the generated profile

resolves #218

Signed-off-by: Will Dower <wdower@mitre.org>

### CODE REVIEW

**Comment by aaronlippold** [comment]
I'd use a v3 RedHat xccdf

**Comment by wdower** [comment]
Updated the RHEL7 V1R4 example profile to V3R3

File: lib/inspec_tools/plugin_cli.rb [hunk 1]
```diff
@@ -36,8 +36,9 @@ class CliCommand < Inspec.plugin(2, :cli_command) # rubocop:disable Metrics/Clas
       option :separate_files, required: false, type: :boolean, default: true, aliases: '-s'
       option :replace_tags, type: :array, required: false, aliases: '-r'
       option :metadata, required: false, aliases: '-m'
+      option :control_id, required: false, enum: %w{ruleID vulnID}, aliases: '-c', default: 'vulnID'
       def xccdf2inspec
-        xccdf = InspecTools::XCCDF.new(File.read(options[:xccdf]), options[:replace_tags])
+        xccdf = InspecTools::XCCDF.new(File.read(options[:xccdf]), options[:control_id] == 'vulnID',options[:replace_tags])
```
Comment (rbclark) [inline lib/inspec_tools/plugin_cli.rb:41]: I guess we're not running rubocop on this project, anyway there is a space missing between the comma and options here.

File: test/unit/inspec_tools/xccdf_test.rb [hunk 2]
```diff
@@ -42,8 +42,17 @@ def test_to_inspec_metadata
...
+    assert_equal(249, inspec_json['controls'].count)
+  end
+
+  def test_control_id_flag
+    xccdf = InspecTools::XCCDF.new(File.read('examples/xccdf2inspec/data/U_RHEL_7_STIG_V3R3_Manual-xccdf.xml'), true)
+    vulnID_inspec_json = xccdf.to_inspec
+    xccdf = InspecTools::XCCDF.new(File.read('examples/xccdf2inspec/data/U_RHEL_7_STIG_V3R3_Manual-xccdf.xml'), false)
+    ruleID_inspec_json = xccdf.to_inspec
+    assert(vulnID_inspec_json['controls'].first['id'].start_with? 'V-')
+    assert(ruleID_inspec_json['controls'].first['id'].start_with? 'SV-')
```
Comment (rbclark) [inline test/unit/inspec_tools/xccdf_test.rb:56]: To make it easier to understand this test I'd say this should probably be broken out into 2 different tests, 1 for the true and one for the vulnID and one for the ruleID.

**Review by rbclark — CHANGES_REQUESTED** [review]

File: test/unit/inspec_tools/xccdf_test.rb (hunk 2 above)
Comment (wdower) [reply test/unit/inspec_tools/xccdf_test.rb]: Changes made

File: .rubocop.yml [hunk 3]
```diff
@@ -195,3 +195,5 @@ Layout/InitialIndentation:
   Enabled: true
 Layout/LeadingCommentSpace:
   Enabled: true
+Metrics/AbcSize:
+  Enabled: false
```
Comment (rbclark) [inline .rubocop.yml:199]: This should be in the `.rubocop_todo.yml` and can be generated by running `bundle exec rubocop --auto-gen-config`

**Review by rbclark — CHANGES_REQUESTED** [review]

**Review by Bialogs — COMMENTED** [review]
Why is the option a string instead of boolean?

**Comment by wdower** [comment]
I think we left it a string so that we could eventually use any tag for the control_id as opposed to just a flag for using the legacy vulnID (which for now would need to default to true anyway). But the logic using the variable does expect it to be either ruleID or vulnID.

**Comment by aaronlippold** [comment]
note that there seem to be at least two ids hat are considered 'legecy' so it looks like it need to be an array

**Comment by rbclark** [comment]
> note that there seem to be at least two ids hat are considered 'legecy' so it looks like it need to be an array

We are already mapping over the legacy array so this shouldn't be an issue.

**Review by rbclark — APPROVED** [review]

**Pull request merged**